### PR TITLE
validns-0.8

### PR DESCRIPTION
**Pull request by Habbie**

### CODE REVIEW

File: Library/Formula/validns.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Validns < Formula
+  homepage "http://www.validns.net/"
+  url "http://www.validns.net/download/validns-0.8.tar.gz"
+  sha256 "df2db0eaa998a0411ff4c1c4e417eb82d32aec4835f92f45f26c66c8d1d5bd22"
+
+  depends_on 'judy'
+
+  patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/validns.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Validns < Formula
+  homepage "http://www.validns.net/"
+  url "http://www.validns.net/download/validns-0.8.tar.gz"
+  sha256 "df2db0eaa998a0411ff4c1c4e417eb82d32aec4835f92f45f26c66c8d1d5bd22"
+
+  depends_on 'judy'
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Use double-quotes everywhere (and fix everything else suggested by `brew audit --strict validns`)

File: Library/Formula/validns.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  patch :DATA
+
+  def install
+    system "make"
+    bin.install 'validns'
+    man1.install 'validns.1'
+  end
+
+  test do
+    system "#{bin}/validns -v 2>&1 | grep 'quit on first validation error'"
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/validns.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  sha256 "df2db0eaa998a0411ff4c1c4e417eb82d32aec4835f92f45f26c66c8d1d5bd22"
+
+  depends_on 'judy'
+
+  patch :DATA
+
+  def install
+    system "make"
+    bin.install 'validns'
+    man1.install 'validns.1'
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Is there no `make install` we can use?

File: Library/Formula/validns.rb (hunk 1 above)
Comment (Habbie) [inline Library/Formula/validns.rb]: The patch is no longer needed after depending on brew openssl.

File: Library/Formula/validns.rb (hunk 2 above)
Comment (Habbie) [inline Library/Formula/validns.rb]: Fixing this.

File: Library/Formula/validns.rb (hunk 4 above)
Comment (Habbie) [inline Library/Formula/validns.rb]: No :(

File: Library/Formula/validns.rb (hunk 3 above)
Comment (Habbie) [inline Library/Formula/validns.rb]: Fixing this.

**Comment by Habbie** [comment]
All comments addressed in f8a73b4 I believe.

File: Library/Formula/validns.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    system "make"
+    bin.install "validns"
+    man1.install "validns.1"
+  end
+
+  test do
+    (testpath/"example.com").write("example.com. IN SOA ns.example.com. hostmaster.example.com. 9 10800 3600 604800 3600")
+    system "#{bin}/validns example.com 2>&1 | grep 'ttl not specified and default is not known'"
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Use `assert_match` (a Ruby function) and `shell_output` instead of `grep` and `2>&1`

Comment (Habbie) [inline Library/Formula/validns.rb]: Fixing this like in splint.rb instead

**Comment by Habbie** [comment]
Test updated.

File: Library/Formula/validns.rb (hunk 5 above)
Comment (Habbie) [inline Library/Formula/validns.rb]: hope that's okay :)

File: Library/Formula/validns.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+  def install
+    system "make"
+    bin.install "validns"
+    man1.install "validns.1"
+  end
+
+  test do
+    (testpath/"example.com").write("example.com. IN SOA ns.example.com. hostmaster.example.com. 9 10800 3600 604800 3600")
+    output = `#{bin}/validns example.com 2>&1`
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Use `shell_output` instead and parse something from `stdout` if possible.

File: Library/Formula/validns.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    assert output.include?("ttl not specified and default is not known")
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Use `assert_match` instead

File: Library/Formula/validns.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    assert_equal 1, $?.exitstatus
```
Comment (mikemcquaid) [inline Library/Formula/validns.rb]: Use `shell_output` with `1` as the second argument which does this assertion.

**Comment by Habbie** [comment]
Test updated to use `assert_match` and `shell_output` with `1`

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `validns 0.8 (new formula)`.

**Pull request closed without merging**